src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java

```diff
@@ -1163,4 +1167,43 @@ public static boolean shouldWriteLocks(Project project) {
         return startParameter.isWriteDependencyLocks()
                 || WRITE_VERSIONS_LOCKS_TASK_NAME_MATCHER.matchesAny(startParameter.getTaskNames());
     }
+
+    @SuppressWarnings("deprecation")
+    public abstract static class ProjectDependencyWorkarounds {
+
+        @Inject
+        public abstract Gradle getGradle();
+
+        /**
+         * {@link ProjectDependency#getPath} was introduced in Gradle 8.11, and
+         * {@link ProjectDependency#getDependencyProject} was not deprecated yet. Fallback to using the latter on
+         * versions prior to 8.11.
+         */
+        public String getPath(ProjectDependency projectDependency) {
+            GradleVersion requiredVersion = GradleVersion.version("8.11");
+            if (GradleVersion.current().compareTo(requiredVersion) < 0) {
+                return projectDependency.getDependencyProject().getPath();
+            } else {
+                return projectDependency.getPath();
+            }
+        }
+
+        /**
+         * {@link ProjectDependency#getDependencyProject} was deprecated in Gradle 8.11 and will be removed in Gradle
+         * 9.0. Without a considerable rethinking of how GCV works with Isolated projects, it may be easier to
+         * continue to reach into subprojects at will for now.
+         *
+         */
+        public Project getDependencyProject(ProjectDependency projectDependency) {
+            GradleVersion removedVersion = GradleVersion.version("9.0");
+            if (GradleVersion.current().compareTo(removedVersion) < 0) {
+                return projectDependency.getDependencyProject();
+            }
+
+            return getGradle().getRootProject().project(projectDependency.getPath());
+        }
+
+        @Inject
+        public ProjectDependencyWorkarounds() {}
```

Comment:
Do you need this? I though it was fine with the default constructor?

Reply:
Similarly I thought you could at least make the class/ctor package-private rather than public?

Reply:
okay, if you have package private class, checkstyle will complain at you for a redundant public modifier. So I think just need to settle on `protected` class with `public` constructor.